src/modules/mc_att_control/mc_att_control_main.cpp

```diff
@@ -693,7 +693,7 @@ MulticopterAttitudeControl::control_attitude(float dt)
 		q.from_dcm(R.transposed() * R_sp);
 		math::Vector<3> e_R_d = q.imag();
 		e_R_d.normalize();
-		e_R_d *= 2.0f * atan2f(e_R_d.length(), q(0));
+		e_R_d *= 2.0f * acosf(q(0));
```

Comment:
@einma1 I thought about this again and I was wondering if you could get into a corner case here.
For every physical attitude there are two quaternions which represent it (q and -q).
In this case we would need to observe the sign of q_error(0) otherwise the controller would rotate into the wrong direction (controller unwinding). The code would look like this:

e_R_d *= sign(q(0)) * 2 * acosf(q(0))